mlflow/metrics/utils/metric_definitions.py

```diff
@@ -249,3 +249,102 @@ def _rougeLsum_eval_fn(eval_df, metrics):
             scores=scores,
             aggregate_results=standard_aggregations(scores),
         )
+
+
+def _mae_eval_fn(eval_df, metrics):
+    if "target" in eval_df:
+        from sklearn.metrics import mean_absolute_error
+
+        y_pred = eval_df["prediction"]
+        y = eval_df["target"]
+
+        mae = mean_absolute_error(y, y_pred)
```

Comment:
These are missing `sample_weight` compared to how they are [calculated now in mlflow.evaluate](https://github.com/mlflow/mlflow/blob/7a592885ca4c827e6951e06ebab5fcbd614e56e4/mlflow/models/evaluation/default_evaluator.py#L122) - do we want to add this as an input to eval_fn in make_metric? cc @sunishsheth2009 @prithvikannan

Reply:
Ahh interesting, missed that. This brings us to the point of we need some things from the evaluator_configs to compute the results. Need to think about this on how we can get that information.
cc @dbczumar to get some thoughts here since we explicitly decide to now include evaluator_config in make_metric.

Reply:
In addition to `sample_weight`, there are also parameters for the classification metrics that help indicate whether they are for binary or multiclass classification. If we have a evaluator_config, we may also want to be able to pass in that information.

Reply:
@sunishsheth2009 @jessechancy I think `pos_label` and `sample_weights` should be optional variables defined on the metric. Metrics have the concept of variables, right?